examples/step-90/step-90.cc

```diff
@@ -0,0 +1,1344 @@
+              //  triangulation.
+              typename DoFHandler<dim>::active_cell_iterator level_set_cell(
+                &(triangulation),
+                cell->level(),
+                cell->index(),
+                &level_set_dof_handler);
+              // Once we know what type of quadratures are required for this
+              // stabilization scheme, we reinit() them in the TraceFEM
+              // ScratchData,
+              //  before acquiring the local matrix contribution in the
+              //  CopyData.
+              scratch_data.fe_values.reinit(cell);
+              scratch_data.level_set_fe_values.reinit(level_set_cell);
+              stabilization_scheme.cell_worker_assemble(cell,
+                                                        scratch_data,
+                                                        copy_data);
+            }
+        }
+    };
+    // This copier worker distributes the local contributions from the CopyData
+    // taking into account the constraints.
+    const auto copier = [&](const TraceFEM_CopyData<dim> &c) {
+      constraints.distribute_local_to_global(c.cell_matrix,
+                                             c.cell_rhs,
+                                             c.local_dof_indices,
+                                             global_matrix,
+                                             global_rhs);
+    };
+
+    // Finally, we set up the TraceFEM ScratchData with the discrete level-set,
+    // classifier and proper quadrature rules.
+    TraceFEM_ScratchData<dim> scratch_data(mapping,
+                                           fe_collection,
+                                           mesh_classifier,
+                                           level_set_dof_handler,
+                                           level_set,
+                                           surface_update_flags,
+                                           cell_quadrature,
+                                           face_quadrature,
+                                           quadrature_1D,
+                                           face_update_flags);
+    TraceFEM_CopyData<dim>    copy_data;
+    // The MeshWorker goes over all cells provided by the solutions'
+    // DOF-handler. Note that this includes non-intersected cells as well,
+    //  but the cell worker does nothing on them.
+    MeshWorker::mesh_loop(dof_handler.begin_active(),
+                          dof_handler.end(),
+                          cell_worker,
+                          copier,
+                          scratch_data,
+                          copy_data,
+                          MeshWorker::assemble_own_cells);
+
+    global_matrix.compress(VectorOperation::add);
+    global_rhs.compress(VectorOperation::add);
+
+    timer.stop();
+    pcout << "took (" << timer.wall_time() << "s)" << std::endl;
+  }
+
+  // The straighforward direct solver is called in the following method.
```

Comment:
In the following, we solve the resulting linear system of equations. We either use a direct solver or AMG.